### ISSUE TITLE
Uncaught exception causes a deadlock when `n_jobs` is more than 1.

### ISSUE THREAD

**Issue by g-votte** [optuna/optuna#538]
`study.optimize()` results in a deadlock when `n_jobs` is more than 1 and the error in the objective function is not caught by Optuna.

**Code to reproduce**

```python
import optuna

def objective(trial):
    raise ValueError()

study = optuna.create_study()
study.optimize(objective, n_jobs=2, n_trials=10, catch=())
```

**Comment by sile** [comment]
This problem may be resolved by https://github.com/optuna/optuna/pull/692.

**Comment by sile** [comment]
I confirmed #692 fixed this issue (see #769), so I'll close this PR.
